### PR TITLE
Remove deepcopy

### PR DESCRIPTION
**Pull request by elad-c**

### CODE REVIEW

File: model_compression_toolkit/core/common/substitutions/apply_substitutions.py [hunk 1]
```diff
@@ -13,14 +13,12 @@
...
 
-import copy
-
 from typing import List
 
 from model_compression_toolkit.core import common
 
 
-def substitute(graph_to_substitute: common.Graph,
+def substitute(graph: common.Graph,
```
Comment (ofirgo) [inline model_compression_toolkit/core/common/substitutions/apply_substitutions.py:21]: This one seems a bit dangerous. How can we verify that there in no one substitution that is counting on the fact that the graph is being copied? I'm not sure that our tests can ensure this.

File: model_compression_toolkit/core/runner.py [hunk 2]
```diff
@@ -217,7 +217,7 @@ def get_finalized_graph(initial_graph: Graph,
     ######################################
     # Add quantization configurations
     ######################################
-    transformed_graph = set_quantization_configuration_to_graph(graph=transformed_graph,
+    transformed_graph = set_quantization_configuration_to_graph(graph_with_qcs=transformed_graph,
```
Comment (ofirgo) [inline model_compression_toolkit/core/runner.py:220]: the graph that you are passing here doesn't have QCs, so the name of the argument is confusing.

**Review by ofirgo — CHANGES_REQUESTED** [review]

**Review by ofirgo — APPROVED** [review]

**Comment by haihabi** [comment]
Please link to the issue

File: model_compression_toolkit/core/keras/keras_implementation.py [hunk 3]
```diff
@@ -13,6 +13,7 @@
 # limitations under the License.
 # ==============================================================================
 from typing import List, Any, Tuple, Callable, Type, Dict
+from copy import deepcopy
```
Comment (ofirgo) [inline model_compression_toolkit/core/keras/keras_implementation.py:16]: Is this import necessary?

**Review by ofirgo — APPROVED** [review]

**Pull request merged**